### PR TITLE
Add `GetTokenSupply` and `GetProtocolParameters` to the message inter…

### LINKED ISSUE #1281: Port protocol parameters getter(s) to the bindings
## Description
It would be useful to have equivalents of these methods from the Rust library
```
Client::get_token_supply(&self) -> Result<u64>;
Client::get_protocol_parameters -> Result<ProtocolParameters>; 
```
ported to the bindings (at least from what I could tell they do not exist in Nodejs and Python as of now). 

## Motivation
In identity.rs we need a way for our JS users to obtain these parameters so they can pass them in to our library (which then uses them when calling functionality from say `bee::block` etc.). Note that our Wasm bindings to this library is currently based on a fork where we may implement this ourselves, but it would be better to collaborate on an approach that can work for all the bindings of `iota-client`. 

## Are you planning to do it yourself in a pull request?
I am happy to help, but it may take some time since I have not contributed to this code base before.

### PR DESCRIPTION
**Pull request by Thoralf-M**
…face

## Links to any relevant issues

Fixes https://github.com/iotaledger/iota.rs/issues/1281

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

Running example calling the new methods

## Change checklist

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by samuel-rufi — DISMISSED** [review]

**Review by thibault-martinez — COMMENTED** [review]
Do we really have to provide both `GetTokenSupply` and `GetProtocolParameters` in the message interface ? `GetProtocolParameters` should be enough since the token_supply is in the parameters. If they really want a getter for the supply we can do it but only in TS calling the protocol and returning the supply.

**Comment by Thoralf-M** [comment]
@thibault-martinez they could also get all from https://github.com/iotaledger/iota.rs/issues/1281#issuecomment-1261146978 :man_shrugging:

**Comment by thibault-martinez** [comment]
> @thibault-martinez they could also get all from [#1281 (comment)](https://github.com/iotaledger/iota.rs/issues/1281#issuecomment-1261146978) 🤷‍♂️

Yeah but ProtocolParameters is cached

**Comment by Thoralf-M** [comment]
> Yeah but ProtocolParameters is cached

NetworkInfo is also cached

File: client/bindings/nodejs/lib/Client.ts [hunk 1]
```diff
@@ -421,6 +422,24 @@ export class Client {
...
+     * Returns the token supply.
+     */
+    async getTokenSupply(): Promise<string> {
+        return (await this.getProtocolParameters()).tokenSupply;
+    }
+
+    /**
+     * Returns the protocol parameters.
+     */
+    async getProtocolParameters(): Promise<INodeInfoProtocol> {
```
Comment (thibault-martinez) [inline client/bindings/nodejs/lib/Client.ts:435]: Just checking, is `INodeInfoProtocol` correct here ?

Comment (Thoralf-M) [reply client/bindings/nodejs/lib/Client.ts:435]: Yes, it's the type from iota.js

**Review by thibault-martinez — APPROVED** [review]

**Review by samuel-rufi — APPROVED** [review]

**Pull request merged**